### DISCUSSION TITLE
Transportation - How to show single attributes from road column in QGIS

### DISCUSSION THREAD

**Discussion by anhe91** [OvertureMaps/data#303] [Discussion]
Hello,

it could be possible that my problem is not special for Overture Data, but I hope you can help me.

When I import the transportation data into QGIS, the whole data scheme of the road attributes show up in a single column "road". 

![image](https://github.com/OvertureMaps/schema/assets/149072588/ec3b928e-d8b5-40e0-b377-9193584cd3ff)

My goal is to display the transportation layer in different colors, in relationship to the speed limit of a segment. My first idea was to search for speed limits in the road column and extract the numbers into a new column. Of course this doesnt work so easy, because multiple speed limits in one segment can be possible and also because of the linear referencing.

Can someone help me how I can solve my problem? I don't know which is the best way....maybe it is possible to just display the data from the given data structure with a query.

Thank you a lot in advance!

Best regards,
André

**Comment by sanjibdutta1** [comment]
Road property is a JSON string, why not use a JSON library to find the speed limits?

**Comment by anhe91** [comment]
I was able to extract the speed limits from the "Road" column to the "speedLimit" column or each individual speed limit to a single column (Speed1, Speed 2, Speed 3) if there are different speed limits in the segment.

Does anyone have an idea how I can separate the segments to the given speed limits?

![image](https://github.com/OvertureMaps/schema/assets/149072588/e0f34f9a-3cec-4508-a89d-17d3f7697445)

**Reply by anhe91** [reply]
I could solve the problem. I splitted the speed limit and its linear reference in separate columns (by search and replace in Excel *Awkward*) and then I used the QGIS extensions Qchainage and LRS to transform the Overture road network and locate the speed limits in the segments.

**Comment by brad-richardson** [comment]
@anhe91 we are considering dropping JSON entirely from the segment schema. Any feedback on that? https://github.com/OvertureMaps/schema/discussions/184
